src/bundle/Controller/ContentTypeController.php

```diff
@@ -71,22 +82,43 @@ public function __construct(
         ActionDispatcherInterface $contentTypeActionDispatcher,
         FormFactory $formFactory,
         SubmitHandler $submitHandler,
-        array $languages
+        RequestStack $requestStack,
+        array $languages,
+        int $defaultPaginationLimit
     ) {
         $this->notificationHandler = $notificationHandler;
         $this->translator = $translator;
         $this->contentTypeService = $contentTypeService;
         $this->contentTypeActionDispatcher = $contentTypeActionDispatcher;
         $this->formFactory = $formFactory;
         $this->submitHandler = $submitHandler;
+        $this->requestStack = $requestStack;
         $this->languages = $languages;
+        $this->defaultPaginationLimit = $defaultPaginationLimit;
     }
 
+    /**
+     * @param ContentTypeGroup $group
+     *
+     * @return Response
+     */
     public function listAction(ContentTypeGroup $group): Response
     {
+        $parentRequest = $this->requestStack->getParentRequest();
+
         $deletableTypes = [];
 
-        $types = $this->contentTypeService->loadContentTypes($group, $this->languages);
+        $page = $parentRequest->query->get('page') ?? 1;
```

Comment:
Why can't you just define `$page` as a part of the route so you don't have to access `RequestStack` as this feels dirty, really dirty?

Reply:
It's not only about page. I need also route name from parent request. I think this is a better approach than pass this variables in `    {{ render(controller('EzPlatformAdminUiBundle:ContentType:list', {
        contentTypeGroupId: content_type_group.id
    })) }}`

Or maybe this should be standalone action instead render controller?